Fix issue when using resin deploy with non-standard stdin

pimterry opened the pull request:
Connects to #539

With non-TTY inputs, and other weird stdin cases (e.g. git bash) functions like `process.stdin.clearLine()` are not defined. The resin-cli-visuals progress bar doesn't depend on these, so this PR moves to use that instead to fix #539, which currently breaks `resin deploy` for all git bash users on windows.

This also lets us neatly gives us an ETA for the image push too, for free (it's already being calculated by progress-stream).

Output looks like:

```bash
[Build]   Using cache
[Build]    ---> 5a029c04a533
[Build]   Successfully built 5a029c04a533

...unicorn...

[Info]    Initializing deploy...
Deploying [============            ] 50% eta 36s  
```

One downside is that this now doesn't include the coloured `[Info]` prefix. @CameronDiver how important is that? I could try to extend resin-cli-visuals to accept  arbitrary streams to write to, rather than using console.log, but it's a little fiddly, and most other commands don't use this log streaming anyway (and neither does the unicorn output). Thoughts?

CameronDiver commented on lib/actions/deploy.coffee:
```diff
@@ -31,30 +31,17 @@ parseInput = Promise.method (params, options) ->
-	ansiEscapes = require('ansi-escapes')
+showPushProgress = ->
+	visuals = require('resin-cli-visuals')
+	progressBar = new visuals.Progress('Deploying')
```
If we did want to add prefixes, this function could be used: https://github.com/resin-io-modules/stream-logger/blob/master/src/index.ts#L53, and maybe could switch to `[Success]` when completed.

dfunckt commented on lib/actions/deploy.coffee:
```diff
@@ -31,30 +31,17 @@ parseInput = Promise.method (params, options) ->
+	progressBar.update({ percentage: 0 })
+	return progressBar
 
+updatePushProgress = (percentage, eta, progressBar) ->
```
I think this is unused now?

emirotin commented on package.json:
```diff
@@ -71,7 +71,7 @@
     "resin-cli-errors": "^1.2.0",
     "resin-cli-form": "^1.4.1",
-    "resin-cli-visuals": "^1.3.0",
+    "resin-cli-visuals": "^1.4.0",
```
Needs rebasing, it's already on master

pimterry commented:
Sorry that took me a little while - this is now updated to add a prefix following @CameronDiver's suggestion. Output now looks like:

```
[Info]    Initializing deploy...
[Info]    Deploying [========================] 100% eta 0s            
[Info]    Remote: Accepting image for application picam, owner: gh_pimterry
```

Doesn't replace the progress bar with success, both because that's not what the builder does and because it's fiddly, but it's now at least a bit tidier and more consistent.

Rereview please - notably to make this happen I've refactored the logStreams code substantially, so that we can just pass around a logger object, rather than using the streams directly (because that made it impossible to get at the formatting logic).
